### ISSUE TITLE
io.zeebe.broker.it.clustering.topology.TopologyClusterTest is flaky

### ISSUE THREAD

**Issue by Zelldon** [zeebe-io/zeebe#3216]
**Describe the bug**
Failed now multiple times in the CI
```
[ERROR] io.zeebe.broker.it.clustering.topology.TopologyClusterTest  Time elapsed: 0.006 s  <<< FAILURE!

java.lang.AssertionError: Failed to wait for topology TopologyImpl{brokers=[BrokerInfoImpl{nodeId=0, host='0.0.0.0', port=26802, partitions=[]}, BrokerInfoImpl{nodeId=1, host='0.0.0.0', port=26807, partitions=[]}, BrokerInfoImpl{nodeId=2, host='0.0.0.0', port=26812, partitions=[]}], clusterSize=3, partitionsCount=3, replicationFactor=3}

	at io.zeebe.test.util.TestUtil$Invocation.until(TestUtil.java:133)

	at io.zeebe.test.util.TestUtil$Invocation.until(TestUtil.java:81)

	at io.zeebe.test.util.TestUtil.waitUntil(TestUtil.java:53)

	at io.zeebe.broker.it.clustering.ClusteringRule.waitForTopology(ClusteringRule.java:533)

	at io.zeebe.broker.it.clustering.ClusteringRule.waitForPartitionReplicationFactor(ClusteringRule.java:388)

	at io.zeebe.broker.it.clustering.ClusteringRule.before(ClusteringRule.java:184)

	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:46)

	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:298)

	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:292)

	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)

	at java.base/java.lang.Thread.run(Thread.java:834)
```

**Comment by npepinpe** [comment]
I think this should be fixed by #3267; locally I can't reproduce it anymore after running > 100 times. Feel free to reopen if it happens again.

**Comment by Zelldon** [comment]
Still fails from time to time:

```
TopologyClusterTest.io.zeebe.broker.it.clustering.topology.TopologyClusterTest Failed to wait for topology TopologyImpl{brokers=[BrokerInfoImpl{nodeId=1, host='0.0.0.0', port=26807, partitions=[PartitionInfoImpl{partitionId=2, role=FOLLOWER}, PartitionInfoImpl{partitionId=3, role=FOLLOWER}, PartitionInfoImpl{partitionId=1, role=FOLLOWER}]}, BrokerInfoImpl{nodeId=2, host='0.0.0.0', port=26812, partitions=[PartitionInfoImpl{partitionId=1, role=FOLLOWER}]}, BrokerInfoImpl{nodeId=0, host='0.0.0.0', port=26802, partitions=[PartitionInfoImpl{partitionId=2, role=FOLLOWER}, PartitionInfoImpl{partitionId=3, role=FOLLOWER}]}], clusterSize=3, partitionsCount=3, replicationFactor=3}
```

**Comment by npepinpe** [comment]
Could this have been related to the fix @deepthidevaki did (#3708)? I checked recent PRs from the past week and it stopped happening - I also cannot reproduce it locally. If nobody has any objections, I'll mark as tentatively fixed for now.

**Comment by npepinpe** [comment]
FYI, this still happens (the error itself), but seems unrelated to the `TopologyClusterTest`, it happens every once in a while across various tests (as part of the `ClusteringRule`), so I'll open another issue instead of this one.
